src/robot/parsing/model/statements.py

```diff
@@ -355,42 +577,152 @@ def name(self):
+        tokens = [Token(Token.SEPARATOR, indent), Token(Token.SETUP, '[Setup]')]
+        if keyword:
+            tokens.append(Token(Token.SEPARATOR, separator))
+            tokens.append(Token(Token.NAME, keyword))
+        if args is not None:
+            for arg in args:
+                tokens.append(Token(Token.SEPARATOR, separator))
+                tokens.append(Token(Token.ARGUMENT, arg))
+        tokens.append(Token(Token.EOL, eol))
+        return cls(tokens)
+
 
 @Statement.register
 class Teardown(Fixture):
     type = Token.TEARDOWN
 
+    @classmethod
+    def from_params(cls, keyword, args=None, indent=FOUR_SPACES, separator=FOUR_SPACES, eol=EOL):
+        tokens = [Token(Token.SEPARATOR, indent), Token(Token.TEARDOWN, '[Teardown]')]
+        if keyword:
+            tokens.append(Token(Token.SEPARATOR, separator))
+            tokens.append(Token(Token.NAME, keyword))
+        if args is not None:
+            for arg in args:
+                tokens.append(Token(Token.SEPARATOR, separator))
+                tokens.append(Token(Token.ARGUMENT, arg))
+        tokens.append(Token(Token.EOL, eol))
+        return cls(tokens)
+
 
 @Statement.register
 class Tags(MultiValue):
     type = Token.TAGS
 
+    @classmethod
+    def from_params(cls, tags, indent=FOUR_SPACES, separator=FOUR_SPACES, eol=EOL):
+        tokens = [
+            Token(Token.SEPARATOR, indent),
+            Token(Token.TAGS, '[Tags]')
+        ]
+
+        for tag in tags:
+            tokens.append(Token(Token.SEPARATOR, separator))
+            tokens.append(Token(Token.ARGUMENT, tag))
+        tokens.append(Token(Token.EOL, eol))
+        return cls(tokens)
+
 
 @Statement.register
 class Template(SingleValue):
     type = Token.TEMPLATE
 
+    @classmethod
+    def from_params(cls, keyword=None, indent=FOUR_SPACES, separator=FOUR_SPACES, eol=EOL):
+        tokens = [
+            Token(Token.SEPARATOR, indent),
+            Token(Token.TEMPLATE, '[Template]')
+        ]
+        if keyword:
+            tokens.append(Token(Token.SEPARATOR, separator))
+            tokens.append(Token(Token.NAME, keyword))
+        tokens.append(Token(Token.EOL, eol))
+        return cls(tokens)
+
 
 @Statement.register
 class Timeout(SingleValue):
     type = Token.TIMEOUT
 
+    @classmethod
+    def from_params(cls, timeout=None, indent=FOUR_SPACES, separator=FOUR_SPACES, eol=EOL):
+        tokens = [
+            Token(Token.SEPARATOR, indent),
+            Token(Token.TIMEOUT, '[Timeout]')
+        ]
+        if timeout:
+            tokens.append(Token(Token.SEPARATOR, separator))
+            tokens.append(Token(Token.ARGUMENT, timeout))
+        tokens.append(Token(Token.EOL, eol))
+        return cls(tokens)
+
 
 @Statement.register
 class Arguments(MultiValue):
     type = Token.ARGUMENTS
 
+    @classmethod
+    def from_params(cls, args, split_to_multiline=False, indent=FOUR_SPACES, separator=FOUR_SPACES, eol=EOL):
```

Comment:
I'm going to add documentation (especially to every method that have some control flag). Agreed on rest. 
Splitting arguments functionality could be useful (I'm already doing it in some cases in some tools) but I'm still not 100% if I should add this to core library. I will think about it.

Reply:
Decided to remove splitting. There is too many possiblities to cover (for example, split every line vs split after x arguments, first argument should be the same line or the next..). If someone needs something more advanced it will be better to use `from_tokens` instead.

Reply:
I was also thinking that there are multiple ways to split args. Handling them is Tidy functionality.